libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -192,16 +267,67 @@ void AP_BattMonitor_UAVCAN::read()
     _state.healthy = _interim_state.healthy;
     _state.time_remaining = _interim_state.time_remaining;
     _state.has_time_remaining = _interim_state.has_time_remaining;
-    _state.is_powering_off = _interim_state.is_powering_off;
+    set_powered_state(_interim_state.powered_state);
+
     memcpy(_state.cell_voltages.cells, _interim_state.cell_voltages.cells, sizeof(_state.cell_voltages));
 
     _has_temperature = (AP_HAL::millis() - _state.temperature_time) <= AP_BATT_MONITOR_TIMEOUT;
 }
 
+void AP_BattMonitor_UAVCAN::set_hardware_to_powered_state(const AP_BattMonitor::PoweredState desired_state)
+{
+    if (desired_state != AP_BattMonitor::PoweredState::Powered_ON && desired_state != AP_BattMonitor::PoweredState::Powered_OFF) {
+        return;
+    }
+
+    const bool power_on = (desired_state == AP_BattMonitor::PoweredState::Powered_ON);
+    if (_is_mppt_packetDigital) {
+        mppt_send_enable_output(power_on);
+    }
+}
+
+void AP_BattMonitor_UAVCAN::mppt_send_enable_output(const bool enable)
+{
+    if (_ap_uavcan == nullptr || _node == nullptr) {
+        return;
+    }
+
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Battery %u: Setting power to: %s", (unsigned)_instance+1, enable?"ON":"OFF");
+
+    mppt::OutputEnable::Request request;
+    request.enable = enable ? true : false;
+    request.disable = !request.enable;
+
+    uavcan::ServiceClient<mppt::OutputEnable> client(*_node);
+    client.setCallback([](const uavcan::ServiceCallResult<mppt::OutputEnable>& handle_mppt_enable_output_response){});
+    client.call(_node_id, request);
+}
+
+void AP_BattMonitor_UAVCAN::handle_mppt_enable_output_response(const uavcan::ServiceCallResult<mppt::OutputEnable>& response)
+{
+    // TODO
+    //const mppt::OutputEnable::Response r = response.getResponse();
```

Comment:
normally we try not to leave commented out code in there..

Reply:
How about an # if 0?

Reply:
Are we really ever going to support it?  If the commented out code is going to sit there forever let's just remove it and add an "unsupported" comment or something similar.

Reply:
replaced with "Not Supported"